zebrad/tests/config.rs

```diff
@@ -0,0 +1,342 @@
+
+    let result = ZebradConfig::load(Some(config_path));
+    assert!(result.is_err(), "Should fail to load invalid TOML");
+}
+
+#[test]
+fn config_invalid_env_values_error() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_RPC__LISTEN_ADDR", "invalid_address");
+
+    let result = ZebradConfig::load(None);
+    assert!(
+        result.is_err(),
+        "Should fail with invalid RPC listen address"
+    );
+}
+
+#[test]
+fn config_nested_env_vars() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_TRACING__FILTER", "debug");
+
+    let config = ZebradConfig::load(None).expect("load config with nested env vars");
+
+    assert_eq!(config.tracing.filter.as_ref().unwrap(), "debug");
+}
+
+// --- Specific env mappings used in Docker examples ---
+
+#[test]
+fn zebra_network_network_env() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_NETWORK__NETWORK", "Testnet");
+
+    let config = ZebradConfig::load(None).expect("load config with ZEBRA_NETWORK__NETWORK");
+    assert_eq!(config.network.network.to_string(), "Testnet");
+}
+
+#[test]
+fn zebra_rpc_listen_addr_env() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_RPC__LISTEN_ADDR", "127.0.0.1:18232");
+
+    let config = ZebradConfig::load(None).expect("load config with ZEBRA_RPC__LISTEN_ADDR");
+    assert_eq!(
+        config.rpc.listen_addr.unwrap().to_string(),
+        "127.0.0.1:18232"
+    );
+}
+
+#[test]
+fn zebra_state_cache_dir_env() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_STATE__CACHE_DIR", "/test/cache");
+
+    let config = ZebradConfig::load(None).expect("load config with ZEBRA_STATE__CACHE_DIR");
+    assert_eq!(config.state.cache_dir, PathBuf::from("/test/cache"));
+}
+
+#[test]
+fn zebra_metrics_endpoint_addr_env() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_METRICS__ENDPOINT_ADDR", "0.0.0.0:9999");
+
+    let config = ZebradConfig::load(None).expect("load config with ZEBRA_METRICS__ENDPOINT_ADDR");
+    assert_eq!(
+        config.metrics.endpoint_addr.unwrap().to_string(),
+        "0.0.0.0:9999"
+    );
+}
+
+#[test]
+fn zebra_tracing_log_file_env() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_TRACING__LOG_FILE", "/test/zebra.log");
+
+    let config = ZebradConfig::load(None).expect("load config with ZEBRA_TRACING__LOG_FILE");
+    assert_eq!(
+        config.tracing.log_file.as_ref().unwrap(),
+        &PathBuf::from("/test/zebra.log")
+    );
+}
+
+#[test]
+fn zebra_mining_miner_address_from_toml() {
+    let _env = EnvGuard::new();
+
+    let miner_address = "u1cymdny2u2vllkx7t5jnelp0kde0dgnwu0jzmggzguxvxj6fe7gpuqehywejndlrjwgk9snr6g69azs8jfet78s9zy60uepx6tltk7ee57jlax49dezkhkgvjy2puuue6dvaevt53nah7t2cc2k4p0h0jxmlu9sx58m2xdm5f9sy2n89jdf8llflvtml2ll43e334avu2fwytuna404a";
+    let toml_string = format!(
+        "[network]\nnetwork = \"Testnet\"\n\n[mining]\nminer_address = \"{}\"",
+        miner_address
+    );
```

Comment:
Optional/Nitpick: We could use a raw str here to make this more readable.

```suggestion
    let toml_string = format!(
        r#"[network]
        network = "Testnet"
        
        [mining]
        miner_address = "{miner_address}""#,
    );
```